addon/components/x-select.js

```diff
@@ -72,38 +72,90 @@ export default Ember.Component.extend({
   tabindex: 0,
 
   /**
+   * Function for the `on-blur` action
+   *
+   * @property on-blur
+   * @type Function
+   */
+  "on-blur": Ember.K,
+
+  /**
+   * Function for the `on-click` action
+   *
+   * @property on-click
+   * @type Function
+   */
+  "on-click": Ember.K,
+
+  /**
+   * Function for the `on-change` action
+   *
+   * @property on-change
+   * @type Function
+   */
+  "on-change": Ember.K,
+
+  /**
+   * Function for the `on-focus-out` action
+   *
+   * @property on-focus-out
+   * @type Function
+   */
+  "on-focus-out": Ember.K,
+
+  /**
+   * Function that calls an action and sends the proper arguments.
+   *
+   * @method _handleAction
+   * @type Function
+   * @param {String} action - string name of the action to invoke
+   * @param {String|Object} value - current value of the component
+   * @param {Object} event - jQuery event from the current action
+   */
+  _handleAction(action, value, event) {
+    let actionValue = this.get(action);
+
+    if(typeof actionValue === 'string') {
+      Ember.warn(`x-select: You must use the action helper for all actions. The try: ${action}=(action "${actionValue}") in your template`, false, {id: 'x-select-string-action'});
+      return;
+    }
+
+    this.get(action)(value, event);
```

Comment:
This is the right track, but I think we can make it a bit clearer and improve the error handling even more.

1. `_handleAction` isn't really what it's doing. It seems like it's dispatching the action so that something _else_ can handle it. `callAction`, `invokeAction`, `doAction` would be better candidates.
1. We issue a warning if `actionValue` is a string, but it if we're going to go to the effort of performing an explicit check on the type of the action value, then we should issue a warning if it's _anything_ except a function. Because anything except a function will fail.
1. If we've already got the action in the form of `actionValue`, then let's just call it:

```js
actionValue(value, event);
```